src/uct/tcp/tcp_ep.c

```diff
@@ -436,7 +436,7 @@ ucs_status_t uct_tcp_ep_create(const uct_ep_params_t *params,
     /* TODO: handle AF_INET6 */
     dest_addr.sin_family = AF_INET;
     dest_addr.sin_port   = *(in_port_t*)params->iface_addr;
-    dest_addr.sin_addr   = *(struct in_addr*)params->dev_addr;
+    dest_addr.sin_addr   = ((struct sockaddr_in*)params->dev_addr)->sin_addr;
```

Comment:
can use the following wrapper:
https://github.com/openucx/ucx/blob/83f089608d57a747543ed829c176c7f345230eb8/src/ucs/sys/sock.h#L417
so, it will be easier to update for AF_INET6 support then
wdyt?

Reply:
@dmitrygx i'll update to use this function, 10x!
please note that since it returns 'void*', i would still have to cast it to (struct in_addr*) which is IPv4.

Reply:
@alinask I see, thank you!